decidim-core/app/middleware/decidim/current_component.rb

```diff
@@ -22,12 +22,17 @@ def call
     private
 
     def detect_current_component(params)
-      return unless params[:current_component_id]
+      return nil unless params[:current_component_id]
 
       organization = @env["decidim.current_organization"]
-      participatory_process = organization.participatory_processes.find(params[:participatory_process_id])
 
-      participatory_process.components.find(params[:current_component_id])
+      participatory_process = organization.participatory_processes.find_by(
```

Comment:
Same thing: We don't have the `request` object before the router, so we can't get the `participatory_process`.